packages/app/src/cli/services/dev.test.ts

```diff
@@ -0,0 +1,70 @@
+import {devNonPreviewableExtensionTarget} from './dev.js'
+import {setupConfigWatcher, setupNonPreviewableExtensionBundler} from './dev/extension/bundler.js'
+import {testApp, testUIExtension} from '../models/app/app.test-data.js'
+import {loadLocalUIExtensionsSpecifications} from '../models/extensions/specifications.js'
+import {describe, expect, test, vi} from 'vitest'
+import {AbortController} from '@shopify/cli-kit/node/abort'
+import {Writable} from 'node:stream'
+
+vi.mock('./dev/extension/bundler.js')
+
+describe('devNonPreviewableExtensionTarget()', () => {
+  test('calls setupNonPreviewableExtensionBundler and setupConfigWatcher on each extension', async () => {
+    const abortController = new AbortController()
+    const stdout = new Writable()
+    const stderr = new Writable()
+    const extension1 = await testUIExtension({
+      devUUID: '1',
+      directory: 'directory/path/1',
+    })
+
+    const extension2 = await testUIExtension({
+      devUUID: '2',
+      directory: 'directory/path/2',
+    })
```

Comment:
could you also add here an extension with isPreviewable: true, so that we test that it is not added to the non-previewable target?